react_main/src/pages/Play/Decks/CreateDeck.jsx

```diff
@@ -1,114 +1,286 @@
+  const params = new URLSearchParams(location.search);
+  const siteInfo = useContext(SiteInfoContext);
+  const errorAlert = useErrorAlert();
+  
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    watch,
+    setValue,
+    getValues,
+    control
+  } = useForm({
+    defaultValues: {
+      defProfile: [{ name: "test", avatar: "", color: "", /*deathMessage: "",*/ id: "", deck: "", preview: "", image: "" }],
     },
-  ]);
+  });
+  const { fields, append, remove } = useFieldArray({
+    name: "cards",
+    control
+  });
+  renderCount++;
 
-  function onCreateDeck(editing) {
-    let profiles = tempParseWordsToProfiles(formFields[1].value);
+  const [deckName, setDeckName] = useState("");
+  const [editing, setEditing] = useState(false);
+  const [selectedFiles, setSelectedFiles] = useState([]);
+
+  const setFile = index => e => {
+    console.log(`index: ${index}`);
+    console.log(`e.target.files: ${e.target.files}`);
+    let newArr = [...selectedFiles]; // copying the old datas array
+    newArr[index] = e.target.files; // replace e.target.value with whatever you want to change it to
+    setSelectedFiles(newArr);
+  }
 
+  if(params.get("edit") && !editing) {
+    setEditing(true);
     axios
-      .post("/deck/create", {
-        name: formFields[0].value,
-        profiles: profiles,
-        editing: editing,
-        id: params.get("edit"),
-      })
-      .then(() => {
-        siteInfo.showAlert(
-          `${editing ? "Edited" : "Created"} deck '${formFields[0].value}'`,
-          "success"
-        );
+      .get(`/deck/${params.get("edit")}`)
+      .then((res) => {
+        let deck = res.data;
+        setDeckName(deck.name);
+        getDeckProfiles(deck.profiles);
       })
       .catch(errorAlert);
   }
 
-  const errorAlert = useErrorAlert();
-  const [editing, setEditing] = useState(false);
+  function removeProfile (index) {
+    remove(index);
+  }
+  
+  function getDeckProfiles(profleIds) {
+    axios
+      .get(`/deck/profiles/${params.get("edit")}`, { params: { ids: profleIds } })
+      .then((res) => {
+        let profiles = res.data;
+        for (let i = 0; i < profiles.length; i++) {
+          profiles[i].image = profiles[i].avatar;
+        }
+        setValue("cards", profiles, { shouldValidate: true, shouldDirty: true });
+      })
+      .catch(errorAlert);
+  }
 
-  const location = useLocation();
-  const params = new URLSearchParams(location.search);
+  function onCreateDeck(editing, data) {
+    let profiles = data.cards;
 
-  const siteInfo = useContext(SiteInfoContext);
+    if (!editing) {
+      axios
+        .post("/deck/create", {
+          name: deckName,
+          profiles: profiles,
+          editing: editing,
+          id: params.get("edit"),
+        })
+        .then((res) => {
+          siteInfo.showAlert(
+            `${editing ? "Edited" : "Created"} deck '${deckName}'`, "success");
+          !editing ? setEditing(true) : setEditing(false);
```

Comment:
editing is definitely false here

Reply:
I don't know what you mean.

Reply:
this part of the code is nested under if (!editing) , line 83